tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -509,6 +515,50 @@ Status IrEmitter::EmitAtomicOperationForNestedComputation(
                                      source_address);
 }
 
+bool IrEmitter::IsEmittingForAMDGPU() const {
+  llvm::Triple target_triple = llvm::Triple(module_->getTargetTriple());
+  return target_triple.isAMDGPU();
+}
+
+bool IrEmitter::EmitAMDGPUAtomics(
+    llvm::Value* output_address, llvm::Value* source,
+    PrimitiveType element_type) {
+  if (IsEmittingForAMDGPU()) {
```

Comment:
I think this is a bit misleading vs. the function name (it says "emits", but in practice only emits for AMD GPUs). Could we make the fact that we are emitting for AMD GPU a precondition with a CHECK and move the test (IsEmitting..()) to the caller?

Reply:
done.  i think the intent has been clarified.